### ISSUE TITLE
unable to open spark log

### ISSUE THREAD

**Issue by dsblr** [rstudio/sparklyr#679]
Hello Sparklyr team,
I am unable to read spark log, I am using spark 2.0.1 and hadoop 2.6 version
Please help me to resolve the issue.
> spark_log(sc)
Error in open.connection(con, "rb") : cannot open the connection
In addition: Warning message:
In open.connection(con, "rb") :
  cannot open file 'log4j.spark.log': No such file or directory

**Comment by BruceZhaoR** [comment]
same issue when used the latest `sparklyr`.  I encounter the issue #295 before, so I used `devtools` to install the latest `sparklyr`, here is the full warning and errors output:

```java
Error in open.connection(con, "rb") : cannot open the connection
In addition: Warning message:
In open.connection(con, "rb") :
  cannot open file 'log4j.spark.log': No such file or directory
Error: java.lang.IllegalArgumentException: Error while instantiating 'org.apache.spark.sql.hive.HiveSessionState':
	at org.apache.spark.sql.SparkSession$.org$apache$spark$sql$SparkSession$$reflect(SparkSession.scala:981)
	at org.apache.spark.sql.SparkSession.sessionState$lzycompute(SparkSession.scala:110)
	at org.apache.spark.sql.SparkSession.sessionState(SparkSession.scala:109)
	at org.apache.spark.sql.SparkSession$Builder$$anonfun$getOrCreate$5.apply(SparkSession.scala:878)
	at org.apache.spark.sql.SparkSession$Builder$$anonfun$getOrCreate$5.apply(SparkSession.scala:878)
	at scala.collection.mutable.HashMap$$anonfun$foreach$1.apply(HashMap.scala:99)
	at scala.collection.mutable.HashMap$$anonfun$foreach$1.apply(HashMap.scala:99)
	at scala.collection.mutable.HashTable$class.foreachEntry(HashTable.scala:230)
	at scala.collection.mutable.HashMap.foreachEntry(HashMap.scala:40)
	at scala.collection.mutable.HashMap.foreach(HashMap.scala:99)
	at org.apache.spark.sql.SparkSession$Builder.getOrCreate(SparkSession.scala:878)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sparklyr.Invoke$.invoke(invoke.scala:94)
	at sparklyr.StreamHandler$.handleMethodCall(stream.scala:89)
	at sparklyr.StreamHandler$.read(stream.scala:55)
	at sparklyr.BackendHandler.channelRead0(handler.scala:49)
	at sparklyr.BackendHandler.channelRead0(handler.scala:14)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:367)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:353)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:346)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:367)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:353)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:346)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:267)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:367)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:353)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:346)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1294)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:367)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:353)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:911)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:652)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.spark.sql.SparkSession$.org$apache$spark$sql$SparkSession$$reflect(SparkSession.scala:978)
	... 44 more
Caused by: java.lang.IllegalArgumentException: Error while instantiating 'org.apache.spark.sql.hive.HiveExternalCatalog':
	at org.apache.spark.sql.internal.SharedState$.org$apache$spark$sql$internal$SharedState$$reflect(SharedState.scala:169)
	at org.apache.spark.sql.internal.SharedState.<init>(SharedState.scala:86)
	at org.apache.spark.sql.SparkSession$$anonfun$sharedState$1.apply(SparkSession.scala:101)
	at org.apache.spark.sql.SparkSession$$anonfun$sharedState$1.apply(SparkSession.scala:101)
	at scala.Option.getOrElse(Option.scala:121)
	at org.apache.spark.sql.SparkSession.sharedState$lzycompute(SparkSession.scala:101)
	at org.apache.spark.sql.SparkSession.sharedState(SparkSession.scala:100)
	at org.apache.spark.sql.internal.SessionState.<init>(SessionState.scala:157)
	at org.apache.spark.sql.hive.HiveSessionState.<init>(HiveSessionState.scala:32)
	... 49 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.spark.sql.internal.SharedState$.org$apache$spark$sql$internal$SharedState$$reflect(SharedState.scala:166)
	... 57 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.spark.sql.hive.client.IsolatedClientLoader.createClient(IsolatedClientLoader.scala:264)
	at org.apache.spark.sql.hive.HiveUtils$.newClientForMetadata(HiveUtils.scala:366)
	at org.apache.spark.sql.hive.HiveUtils$.newClientForMetadata(HiveUtils.scala:270)
	at org.apache.spark.sql.hive.HiveExternalCatalog.<init>(HiveExternalCatalog.scala:65)
	... 62 more
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: The root scratch dir: /tmp/hive on HDFS should be writable. Current permissions are: ---------
	at org.apache.hadoop.hive.ql.session.SessionState.start(SessionState.java:522)
	at org.apache.spark.sql.hive.client.HiveClientImpl.<init>(HiveClientImpl.scala:192)
	... 70 more
Caused by: java.lang.RuntimeException: The root scratch dir: /tmp/hive on HDFS should be writable. Current permissions are: ---------
	at org.apache.hadoop.hive.ql.session.SessionState.createRootHDFSDir(SessionState.java:612)
	at org.apache.hadoop.hive.ql.session.SessionState.createSessionDirs(SessionState.java:554)
	at org.apache.hadoop.hive.ql.session.SessionState.start(SessionState.java:508)
	... 71 more
```

```r
> sc
$master
[1] "local"

$method
[1] "shell"

$app_name
[1] "sparklyr"

$config
$config$sparklyr.cores.local
[1] 4

$config$spark.sql.shuffle.partitions.local
[1] 4

$config$spark.env.SPARK_LOCAL_IP.local
[1] "127.0.0.1"

$config$sparklyr.csv.embedded
[1] "^1.*"

$config$`sparklyr.shell.driver-class-path`
[1] ""

attr(,"config")
[1] "default"
attr(,"file")
[1] "D:\\R\\R-3.4.0\\library\\sparklyr\\conf\\config-template.yml"

$spark_home
[1] "C:\\Users\\Bruce\\AppData\\Local\\rstudio\\spark\\Cache\\spark-2.1.0-bin-hadoop2.7"

$backend
A connection with                               
description "->localhost:61046"
class       "sockconn"         
mode        "wb"               
text        "binary"           
opened      "opened"           
can read    "yes"              
can write   "yes"              

$monitor
A connection with                              
description "->localhost:8880"
class       "sockconn"        
mode        "rb"              
text        "binary"          
opened      "opened"          
can read    "yes"             
can write   "yes"             

$output_file
[1] "C:\\Users\\Bruce\\AppData\\Local\\Temp\\RtmpWooSMo\\file274c63237fb_spark.log"

$spark_context
<jobj[5]>
  class org.apache.spark.SparkContext
  org.apache.spark.SparkContext@36098ded

$java_context
<jobj[6]>
  class org.apache.spark.api.java.JavaSparkContext
  org.apache.spark.api.java.JavaSparkContext@23dae85d

attr(,"class")
[1] "spark_connection"       "spark_shell_connection"
[3] "DBIConnection"  


Session info ------------------------------------------------------------
 setting  value                         
 version  R version 3.4.0 (2017-04-21)  
 system   x86_64, mingw32               
 ui       RStudio (1.0.143)             
 language (EN)                          
 collate  Chinese (Simplified)_China.936
 tz       Asia/Shanghai                 
 date     2017-05-15                    

Packages ----------------------------------------------------------------
 package    * version    date       source                           
 assertthat   0.2.0      2017-04-11 CRAN (R 3.3.3)                   
 backports    1.0.5      2017-01-18 CRAN (R 3.3.2)                   
 base       * 3.4.0      2017-04-21 local                            
 base64enc    0.1-3      2015-07-28 CRAN (R 3.3.0)                   
 compiler     3.4.0      2017-04-21 local                            
 config       0.2        2016-08-02 CRAN (R 3.3.2)                   
 curl         2.6        2017-04-27 CRAN (R 3.4.0)                   
 data.table   1.10.4     2017-02-01 CRAN (R 3.3.2)                   
 datasets   * 3.4.0      2017-04-21 local                            
 DBI          0.6-1      2017-04-01 CRAN (R 3.3.3)                   
 dbplyr       0.0.0.9001 2017-05-15 Github (tidyverse/dbplyr@cf6e0a6)
 devtools     1.13.1     2017-05-13 CRAN (R 3.4.0)                   
 digest       0.6.12     2017-01-27 CRAN (R 3.3.2)                   
 dplyr      * 0.5.0.9004 2017-05-15 Github (tidyverse/dplyr@f12d245)               
 git2r        0.18.0     2017-01-01 CRAN (R 3.3.2)                   
 glue         1.0.0      2017-04-17 CRAN (R 3.4.0)                   
 graphics   * 3.4.0      2017-04-21 local                            
 grDevices  * 3.4.0      2017-04-21 local                            
 grid         3.4.0      2017-04-21 local                            
 hms          0.3        2016-11-22 CRAN (R 3.3.2)                   
 htmltools    0.3.6      2017-04-28 CRAN (R 3.4.0)                   
 httpuv       1.3.3      2015-08-04 CRAN (R 3.3.1)                   
 httr         1.2.1      2016-07-03 CRAN (R 3.3.1)                   
 jsonlite     1.4        2017-04-08 CRAN (R 3.3.3)                   
 knitr        1.15.1     2016-11-22 CRAN (R 3.3.2)                   
 lattice      0.20-35    2017-03-25 CRAN (R 3.4.0)                   
 lazyeval     0.2.0      2016-06-12 CRAN (R 3.3.0)                   
 magrittr     1.5        2014-11-22 CRAN (R 3.3.0)                   
 Matrix       1.2-10     2017-04-28 CRAN (R 3.4.0)                   
 memoise      1.1.0      2017-04-21 CRAN (R 3.4.0)                   
 methods    * 3.4.0      2017-04-21 local                            
 mime         0.5        2016-07-07 CRAN (R 3.3.1)                   
 parallel     3.4.0      2017-04-21 local                            
 R6           2.2.1      2017-05-10 CRAN (R 3.4.0)                   
 rappdirs     0.3.1      2016-03-28 CRAN (R 3.3.1)                   
 Rcpp         0.12.10    2017-03-19 CRAN (R 3.3.3)                   
 readr        1.1.0      2017-03-22 CRAN (R 3.3.3)                   
 rlang        0.1.9000   2017-05-15 Github (tidyverse/rlang@c17568e) 
 rprojroot    1.2        2017-01-16 CRAN (R 3.3.2)                   
 rstudioapi   0.6        2016-06-27 CRAN (R 3.3.1)                   
 sparklyr   * 0.5.4-9002 2017-05-15 Github (rstudio/sparklyr@9f34eb5)
 stats      * 3.4.0      2017-04-21 local                            
 stringi      1.1.5      2017-04-07 CRAN (R 3.3.3)                   
 tibble       1.3.0.9006 2017-05-15 Github (tidyverse/tibble@e311218)
 tools        3.4.0      2017-04-21 local                            
 utils      * 3.4.0      2017-04-21 local                            
 withr        1.0.2      2016-06-20 CRAN (R 3.3.1)                                          
 yaml         2.1.14     2016-11-12 CRAN (R 3.3.2)   
```
If you need further information, please at me at anytime, thank you in advanced, regards!

**Comment by BruceZhaoR** [comment]
on Windows same issues :    #497 #295 #633 #456 , please CC @javierluraschi , thanks!

**Comment by javierluraschi** [comment]
Fix commited, please retry with `sparklyr 0.5.6-9002` or newer. You can install using `devtools::install_github("rstudio/sparklyr")`.
